src/target/riscv/riscv.c

```diff
@@ -3061,30 +3061,22 @@ static int riscv_virt2phys(struct target *target, target_addr_t virtual, target_
 			virtual, physical);
 }
 
-static int riscv_read_phys_memory(struct target *target, target_addr_t phys_address,
-			uint32_t size, uint32_t count, uint8_t *buffer)
+static int check_virt_memory_access(struct target *target, target_addr_t address,
+			uint32_t size, uint32_t count)
 {
-	RISCV_INFO(r);
-	return r->read_memory(target, phys_address, size, count, buffer, size);
+	bool crosses_page_boundary = RISCV_PGBASE(address + size * count - 1) != RISCV_PGBASE(address);
```

Comment:
This part also makes an assumption that that pages are 4 KiB large, so it would be good to make a TODO comment about it:

```suggestion
	// TODO: This assumes that size of each page is 4 KiB, which is not necessarily the case.
	bool crosses_page_boundary = RISCV_PGBASE(address + size * count - 1) != RISCV_PGBASE(address);
```

Reply:
Thanks. Addressed.